streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImplJoin.java

```diff
@@ -184,7 +184,6 @@ private void assertUniqueStoreNames(final WindowBytesStoreSupplier supplier,
         }
     }
 
-    @SuppressWarnings("deprecation") // continuing to support Windows#maintainMs/segmentInterval in fallback mode
```

Comment:
The is nothing to be deprecated -- also the comment is confusing as `maintainMs / segmentInterval` are not use in this method